exploration-module/src/main/java/io/edpn/backend/exploration/adapter/web/dto/mapper/RestSystemDtoMapper.java

```diff
@@ -17,16 +17,16 @@
 public class RestSystemDtoMapper implements SystemDtoMapper {
 
     public SystemDto map(System system) {
-        return new RestSystemDto(system.name(), coordinateFromSystem(system), system.eliteId(), system.starClass());
+        return new RestSystemDto(system.getName(), coordinateFromSystem(system), system.getEliteId(), system.getStarClass());
     }
 
     private CoordinateDto coordinateFromSystem(System system) {
-        boolean isFilled = Optional.ofNullable(system.coordinate())
-                .map(Coordinate::x)
+        boolean isFilled = Optional.ofNullable(system.getCoordinate())
+                .map(Coordinate::getX)
                 .isPresent();
 
         if (isFilled) {
-            return new RestCoordinateDto(system.coordinate().x(), system.coordinate().y(), system.coordinate().z());
+            return new RestCoordinateDto(system.getCoordinate().getX(), system.getCoordinate().getY(), system.getCoordinate().getZ());
```

Comment:
Hmm, couldn't this be simply something to the effect of:
```
return Optional.ofNullable(system.getCoordinate())
    .map(Coordinate::getX)
    .map(coordinate -> new RestCoordinateDto(coordinate.getX(), coordinate.getY(), coordinate.getZ())
    .orElse(null);
```

Reply:
Im not sure the code snippet you showed is accurate.
Optional of Nullable Coordinate, map the x coordinate which spits out a double, map that to a full coordinate?

The issue isnt when coordinate is null, its when coordinate is present but the fields within coordinate are null, so the original code null checks the first coordinate and then if null we dont return and if filled we return values.

Coordinates should either have 3 or none.
However there is nothing to say we dont add a method for a coordinate class that allows us to quickly check that.

I honestly dont have  a strong opinion on how to go about this but currently its both legible and functional so, could be refactored in the future?